### PR TITLE
Add Anexia Cloud Provider Spec to Machine-Controller 

### PR DESCRIPTION
**Pull request by erdii**
**What this PR does / why we need it**: 
This PR adds the cloud provider spec and configs to the machine controller. Next step will be to the full implementation of the provider api in addition to the e2e tests.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Optional Release Note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/cloudprovider/provider/anexia/types/types.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+type RawConfig struct {
+	Token      providerconfigtypes.ConfigVarString `json:"token,omitempty"`
+	VlanID     providerconfigtypes.ConfigVarString `json:"vlanID"`
+	LocationID providerconfigtypes.ConfigVarString `json:"locationID"`
+	TemplateID providerconfigtypes.ConfigVarString `json:"templateID"`
+	Cpus       int                                 `json:"cpus"`
+	Memory     int                                 `json:"memory"`
+	DiskSize   int                                 `json:"diskSize"`
+	SSHKey     providerconfigtypes.ConfigVarString `json:"sshKey"`
```
[the remark this answers is not in the document]
Comment (mfranczy) [reply pkg/cloudprovider/provider/anexia/types/types.go:31]: ok will be removed

File: pkg/cloudprovider/provider/anexia/types/types.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import (
+	providerconfigtypes "github.com/kubermatic/machine-controller/pkg/providerconfig/types"
+)
+
+type RawConfig struct {
+	Token      providerconfigtypes.ConfigVarString `json:"token,omitempty"`
+	VlanID     providerconfigtypes.ConfigVarString `json:"vlanID"`
+	LocationID providerconfigtypes.ConfigVarString `json:"locationID"`
+	TemplateID providerconfigtypes.ConfigVarString `json:"templateID"`
+	Cpus       int                                 `json:"cpus"`
```
[the remark this answers is not in the document]
Comment (mfranczy) [reply pkg/cloudprovider/provider/anexia/types/types.go:28]: ok

File: pkg/cloudprovider/provider/anexia/types/types.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+limitations under the License.
+*/
+
+package types
+
+import (
+	providerconfigtypes "github.com/kubermatic/machine-controller/pkg/providerconfig/types"
+)
+
+type RawConfig struct {
```
[the remark this answers is not in the document]
Comment (zreigz) [reply pkg/cloudprovider/provider/anexia/types/types.go:23]: @moadqassem  just for curiosity. For other providers, we have everywhere `RawConfig`, why you want to change it?

File: pkg/cloudprovider/provider/anexia/types/types.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (mfranczy) [reply pkg/cloudprovider/provider/anexia/types/types.go]: done

File: pkg/cloudprovider/provider/anexia/types/types.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (mfranczy) [reply pkg/cloudprovider/provider/anexia/types/types.go]: done

File: pkg/cloudprovider/provider/anexia/types/types.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (moadqassem) [reply pkg/cloudprovider/provider/anexia/types/types.go:23]: Correct! let's leave it as RawConfig

**Pull request merged**